chispa/default_formats.py

```diff
@@ -1,9 +1,67 @@
-from dataclasses import dataclass
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass, field
+from typing import Any
+
+from chispa.formatting import Color, Format, FormattingConfig, Style
 
 
 @dataclass
 class DefaultFormats:
-    mismatched_rows = ["red"]
-    matched_rows = ["blue"]
-    mismatched_cells = ["red", "underline"]
-    matched_cells = ["blue"]
+    """
+    This class is now deprecated and should be removed in a future release, together with `convert_to_formatting_config`.
+    """
+
+    mismatched_rows: list[str] = field(default_factory=lambda: ["red"])
+    matched_rows: list[str] = field(default_factory=lambda: ["blue"])
+    mismatched_cells: list[str] = field(default_factory=lambda: ["red", "underline"])
+    matched_cells: list[str] = field(default_factory=lambda: ["blue"])
+
+    def __post_init__(self):
+        warnings.warn(
+            "DefaultFormats is deprecated. Use `chispa.formatting.FormattingConfig` instead.", DeprecationWarning
+        )
+
+
+def convert_to_formatting_config(instance: Any) -> FormattingConfig:
+    """
+    Converts an instance of an arbitrary class with specified fields to a FormattingConfig instance.
+    This class is purely for backwards compatibility and should be removed in a future release.
+    """
+
+    if type(instance) is not DefaultFormats:
+        warnings.warn(
+            "Using an arbitrary dataclass is deprecated. Use `chispa.formatting.FormattingConfig` instead.",
+            DeprecationWarning,
+        )
+
+    def _convert_to_format(values: list[str]) -> Format:
```

Comment:
I would like to have it as a private static method of `Format`

Reply:
Good suggestion! Created class method `Format.from_list()`  as alternative constructor.